### ISSUE TITLE
Inference image failed

### ISSUE THREAD

**Issue by joyqoo** [hyBlue/FSRE-Depth#10]
Hi, I used your architecture to train my own dataset, but the results are not as expected, the evaluation results and images are quite poor.

Illogical oddities occur when inferring images. I see that you do not have the inference part.

Could you please give me some advice? Thank you !

<img width="638" alt="166509721-86023852-608a-45e8-97b0-667fdd45d27e" src="https://user-images.githubusercontent.com/30586145/166960358-ea225e26-f28c-45f3-af8a-c5c5151d0a7b.png">
<img width="592" alt="166509320-e34d449b-b34a-4b7b-984d-aa5c4cf23210" src="https://user-images.githubusercontent.com/30586145/166960367-a3197860-90f8-429f-bffe-7d9fbd31ad82.png">

**Comment by hyBlue** [comment]
Hi, 
Are you using night-time images for training? Basically, most self-supervised models may not perform well on night-time images since they are mainly trained with multi-view consistency. 
Also, FSRE performance depends on the segmentation label so you'd better check if the pre-trained segmentation model produced reasonable results on your dataset.  

By the way, I want to ask you which images are FSRE's in your figures. Is it the first column or the last column? And what do you mean by illogical oddities and that it doesn't have the inference part? (You can run inference as in the evaluate.py). 

I haven't tried it on night-time images so I'm not sure about its performance on those. I'd like to see your results and please share them if you don't mind. 

Thanks

**Comment by joyqoo** [comment]
Hi, I'm training with nighttime images of the Oxford Robotcar
The following is the segmented image generated by the segmentation model. I have tried other semantic segmentation models for night time.
But the results are not as clear as NVIDIA's results. Below are the night images and segmentation images.

![001487](https://user-images.githubusercontent.com/30586145/167281644-a26553f9-e566-49e0-9f6c-12d3dc40e3e8.png)
![001487](https://user-images.githubusercontent.com/30586145/167281651-e348afc9-072c-46a9-8810-d4d5d908ade9.png)

My inference image doesn't have any depth representation, it's yellow, and I'm not sure what I should do to improve it.

I can make an evaluation but can't get an inference image, but my evaluation results are poor, as in the previous table.

It would be great if you have any suggestions, thanks for your reply.

**Comment by hyBlue** [comment]
Hi, I'm sorry for the late.
 
What I understood is that 'Ours' in your table shows weird results and it is based on FSRE, but you modified some components in the original FSRE's code other than only the dataset. Is it right?

I'm not sure currently as I don't know about your modifications. However, you need to check whether the training is done properly or not. If the training was well but only the evaluation was wrong, then I think you should check if the models are in 'eval' mode during the evaluation.  The scores seem completely weird and in my experience, those situations sometimes occur due to batchnorm layers.
